### DISCUSSION TITLE
Request for MEDIUMINT Support

### DISCUSSION THREAD

**Discussion by confessore** [diesel-rs/diesel#3536] [Ideas]
I have found a closed issue where it was decided that the mysql MEDIUMINT will not be supported due to there existing no 3byte datatype that would map out to being something like a u24/i24.

on attempting to print the schema from an existing database, encountering the MEDIUMINT type will panic

```
confessore@dev0 my-project% RUST_BACKTRACE=1 diesel print-schema  > src/schema.rs
Encountered unknown type for Mysql: mediumint
thread 'main' panicked at 'internal error: entered unreachable code: Mysql only supports a closed set of types.
                         If you ever see this error message please open an issue at https://github.com/diesel-rs/diesel containing a dump of your schema definition.', /Users/confessore/.cargo/registry/src/github.com-1ecc6299db9ec823/diesel_cli-2.0.1/src/print_schema.rs:313:17
stack backtrace:
   0: rust_begin_unwind
             at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/panicking.rs:584:5
   1: core::panicking::panic_fmt
             at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/panicking.rs:142:14
   2: core::panicking::unreachable_display
   3: <diesel::print_schema::CustomTypeList as core::fmt::Display>::fmt
   4: core::fmt::write
             at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/fmt/mod.rs:1198:17
   5: diesel::print_schema::output_schema
   6: diesel::print_schema::run_print_schema
   7: diesel::main

```

I request we revisit this and provide support for it

**Comment by weiznich** [comment]
I'm happy to accept a PR that adds support for this specific SQL type. This shouldn't include any mapping to rust types for the reasons outlined in the closed issue. If really required such a mapping can be provided by a third party crate.

**Comment by confessore** [comment]
> I'm happy to accept a PR that adds support for this specific SQL type. This shouldn't include any mapping to rust types for the reasons outlined in the closed issue. If really required such a mapping can be provided by a third party crate.

super cool!


i imagine bit shifting could be utilized to maintain a u24/i24 datatype directly in the diesel repo.

i believe the most efficient approach at the moment is to provision a third party crate such as https://crates.io/crates/ux

the contributors of the ux crate have already initiated and persisted much of the brain-energy and work behind this particular subject and i am of the opinion that we should not reinvent this specific wheel

**Reply by weiznich** [reply]
To repeat again what I've already wrote above: I accept a PR that adds a zero sized `diesel::sql_types::MediumInt` type + the necessary `HasSqlType` impls for mysql. I will not accept a PR that adds any `FromSql`/`ToSql` impls to diesel for that type for the reasons discussed in the closed issue. Such impls can for example be added into the linked crate.
